src/system.rs

```diff
@@ -253,6 +272,23 @@ impl PendingTypes {
         type_id + type_store.types_by_id.len()
     }
 
+    /// Adds the [CoreTypeDefinition] and the associated ion_type in the [PendingTypes] and returns the [TypeId] for it
+    /// If the given name already exists in the [TypeStore] or [PendingTypes] it returns the associated [TypeId]
+    pub fn add_core_type(&mut self, ion_type: IonType, type_store: &mut TypeStore) -> TypeId {
+        let core_type = &format!("{:?}", ion_type);
```

Comment:
We probably shouldn't take a direct dependency on whatever format `ion-rust` uses for `IonType`'s `Debug` implementation. While it's not likely to change, it reserves the right to.

Why not make the map a `BTreeMap<IonType, TypeId>`? (We could change it to a `HashMap` after `IonType` is tweaked to derive `Hash`.)

Reply:
I had mainly used the string format because we don't have an implementation of `Hash` for `IonType` to be used with `HashMap` but I can try to use `BTreeMap` and see if it works here.

Reply:
I have tried using `BTreeMap` although it does require `Ord` implmentation for `IonType` which isn't there currently. Hence, I have still kept everything as it is using `String` instead of `IonType` for the `HashMap`. 
Reference issue: #50